python/cudf/cudf/core/dtypes.py

```diff
@@ -166,3 +163,51 @@ def __repr__(self):
             return f"ListDtype({self.element_type.__repr__()})"
         else:
             return f"ListDtype({self.element_type})"
+
+
+class StructDtype(ExtensionDtype):
+
+    name = "struct"
+
+    def __init__(self, fields):
+        """
+        fields : dict
+            A mapping of field names to dtypes
+        """
+        pa_fields = {
+            k: cudf.utils.dtypes.cudf_dtype_to_pa_type(v)
+            for k, v in fields.items()
+        }
+        self._typ = pa.struct(pa_fields)
+
+    @property
+    def fields(self):
+        return {
+            field.name: cudf.utils.dtypes.cudf_dtype_from_pa_type(field.type)
+            for field in self._typ
+        }
+
+    @property
+    def type(self):
+        # TODO: we should change this to return something like a
+        # StructDtypeType, once we figure out what that should look like
+        return dict
+
+    @classmethod
+    def from_arrow(cls, typ):
+        obj = object.__new__(cls)
+        obj._typ = typ
+        return obj
+
+    def to_arrow(self):
+        return self._typ
+
+    def __eq__(self, other):
+        if isinstance(other, str):
+            return other == self.name
+        if type(other) is not StructDtype:
```

Comment:
Should use `isinstance` instead of `type(...)` here. Generally safer, especially if someone subclasses this dtype.

Reply:
Fixed + same for ListsDtype